react/src/components/home-app/Application.jsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import SearchBarComponent from '../search-app/SearchBarComponent';
+
+
+class Application extends React.Component {
+  render() {
+    return (
+      <div>
+        <SearchBarComponent host={ this.props.host } />
+      </div>
+    );
+  }
+}
+
+Application.propTypes = {
+  host: React.PropTypes.string,
```

Comment:
This is minor, but you can specify `isRequired` because by default the props are only optional.

```
host: React.PropTypes.string.isRequired
```

Same for all other components.